model_compression_toolkit/common/mixed_precision/mixed_precision_search_manager.py

```diff
@@ -91,46 +105,74 @@ def get_kpi_metric(self) -> Callable:
-                elif n.is_weights_quantization_enabled():
-                    # The only valid way to get here is if the node is reused (which means that we're not looking
-                    # for its configuration), and we ignore it when computing the KPI (as the base node will acount
-                    # for it).
-                    assert n.reuse, "If node has candidates it should be part of the configurable nodes," \
-                                    " unless it's a reused node"
-                    node_nbits = 0  # Ignore reused nodes is the KPI computation.
-                else:  # No weights quantization
-                    node_nbits = 0
-                node_num_params = 0
-
+                    node_qc = n.candidates_quantization_cfg[mp_model_config[node_idx]]
+                    node_nbits = (node_qc.weights_quantization_cfg.weights_n_bits,
+                                  node_qc.activation_quantization_cfg.activation_n_bits)
+                elif n.is_weights_quantization_enabled() and n.has_weights_to_quantize(self.fw_info):
+                    # The only two valid ways to get here are:
+                    # 1) If the node is reused (which means that we're not looking for its configuration),
+                    #       and we ignore it when computing the KPI (as the base node will account for it).
+                    #       In activation KPI calculation -
+                    #       if we sum all inputs as a metric then we don't want to skip reused nodes.
+                    # 2) If mixed-precision search is only on activation candidates,
+                    #       and weights quantization n_bits is fixed.
+                    assert n.reuse or n.is_all_weights_candidates_equal(), \
+                        "If node has candidates it should be part of the configurable nodes, unless it's a reused " \
+                        "node or the candidates only differ in activation bitwidth"
+                    node_nbits = (0, 0)  # Ignore reused nodes or weights quantization
+                    # only (if no weights mixed-precision) in the KPI computation.
+                else:  # No quantization
+                    node_nbits = (0, 0)
+
+                # Weights memory size computation
                 # Consider only the weights that should be quantized.
-                for attr in self.fw_info.get_kernel_op_attributes(n.type):
-                    if attr is not None:
-                        node_num_params += n.get_weights_by_keys(attr).flatten().shape[0]
-
-                node_memory_in_bytes = node_num_params * node_nbits / 8.0
-                weights_memory += node_memory_in_bytes
-
-            return KPI(weights_memory=weights_memory)
+                if not activation_only and n.is_weights_quantization_enabled() and \
+                        not n.is_all_weights_candidates_equal():
+                    node_num_weights_params = 0
```

Comment:
Because this flag only marks if we want to calculate KPI only for activation/weights, but not the other way around.
activation_only => weights kpi is not computed (do not go into this if).
'not activation_only' does not imply anything.

Reply:
remove the double negation